server/routes/auth.js

```diff
@@ -0,0 +1,121 @@
+const express = require("express");
+const { check, validationResult, cookie } = require("express-validator");
+const UserModel = require("../models/userModel");
+const jwt = require("jsonwebtoken");
+const router = express.Router();
+
+const setJwt = (user) => {
+  const token = jwt.sign({ id: user.id, email: user.email }, process.env.JWT_SECRET, {
+    expiresIn: "14d",
+  });
+
+  return token;
+};
+
+router.post(
+  "/register",
+  [
+    check("name", "Name must not be empty").notEmpty(),
+    check("password", "Must be at least 6 characters long").isLength({
+      min: 6,
+    }),
+    check("email", "Must be an Email").isEmail().notEmpty(),
+  ],
+  async (req, res) => {
+    // if (req.cookies.token) res.redirect('req.headers.host')
+
+    const errors = validationResult(req);
+
+    if (!errors.isEmpty()) {
+      return res.status(400).send(errors);
+    }
+
+    const { name, email, password } = req.body;
+
+    //Check if email is used
+    const exisitingUser = await UserModel.findOne({ where: { email } });
+
+    if (exisitingUser)
+      return res
+        .status(400)
+        .send({ error: { message: "User already exist", code: "400" } });
+
+    const newUser = await UserModel.create({
+      name,
+      email,
+      password,
+    });
+
+    const token = setJwt(newUser);
+
+    //Set token to Cookie
+    res.cookie("token", token);
```

Comment:
It might be a good idea in the future to add some configuration options for the cookie (for security purposes). See [this link](https://expressjs.com/en/api.html#res.cookie) for some possibilities.

Reply:
Yup let's make this an `httpOnly` cookie